app/models/backup_code_configuration.rb

```diff
@@ -0,0 +1,49 @@
+class BackupCodeConfiguration < ApplicationRecord
+  include EncryptableAttribute
+
+  encrypted_attribute_without_setter(name: :code)
+
+  include BackupCodeEncryptedAttributeOverrides
+
+  belongs_to :user
+
+  def self.unused
+    where(used_at: nil)
+  end
+
+  def mfa_enabled?
+    Event.find_by(user_id: user&.id, event_type: :backup_codes_added)
```

Comment:
Minor: might be better with `Event.find_by(user_id: user.id, event_type: :backup_codes_added) if user`

I do wonder though if this would return true if all the codes are used and no new codes are generated.

Would it be better to do something like `!user.backup_code_configurations.all?(:used?)` or `user.backup_code_configurations.unused.any?` (this would mirror the policy)

On the other hand, this might not be used if we rely on the policy object to determine if backup codes are enabled for a user.

Reply:
Got rid of the events based check (was a hack to avoid an old rewrite) and opted for the original user.backup_code_configurations.unused.any?